src/javascript/usage-trends/components/view-selector.js

```diff
@@ -0,0 +1,199 @@
+      property,
+      views,
+      view,
+    });
+  }
+
+  /**
+   * Handles responding to changes in the account <select>.
+   * @param {{target: (Element)}} arg1
+   */
+  handleAccountChange = ({target}) => {
+    const account = this.summaries.getAccount(target.value);
+    this.setSelectedView(account.properties[0].views[0].id);
+  }
+
+  /**
+   * Handles responding to changes in the account <select>.
+   * @param {{target: (Element)}} arg1
+   */
+  handlePropertyChange = ({target}) => {
+    const property = this.summaries.getProperty(target.value);
+    this.setSelectedView(property.views[0].id);
+  }
+
+  /**
+   * Handles responding to changes in the account <select>.
+   * @param {{target: (Element)}} arg1
+   */
+  handleViewChange = ({target}) => {
+    const view = this.summaries.getView(target.value);
+    this.setSelectedView(view.id);
+  }
+
+  /**
+   * React lifecycyle methods below:
+   * http://facebook.github.io/react/docs/component-specs.html
+   * ---------------------------------------------------------
+   */
+
+  /**
+   * Handles querying the Management API for account data the first time the
+   * component is added to the DOM.
+   */
+  componentDidMount() {
+    // The component should not be mounted unless the user is signed in.
+    accountSummaries.get({ignoreEmpty: true}).then((summaries) => {
+      this.summaries = summaries;
+      const accounts = summaries.allAccounts();
+
+      if (!accounts.length) {
+        import('../../analytics').then((analytics) => {
+          analytics.trackNoGoogleAnalyticsAccounts();
+        });
+        AlertDispatcher.addOnce({
+          title: `You don't have any Google Analytics accounts`,
+          message: `To use this tool you need to have read-access to
+                    at least one Google Analytics account.`,
+        });
+        return;
+      }
+
+      if (this.props.viewId) {
+        this.setSelectedView(this.props.viewId);
+      } else {
+        this.setSelectedView(accounts[0].properties[0].views[0].id);
+      }
+    });
+  }
+
+  /**
+   * Handles changes to the viewId prop in parent components.
+   * @param {Object} nextProps
+   */
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.viewId != this.props.viewId) {
+      this.setSelectedView(nextProps.viewId);
+    }
+  }
+
+  /**
+   * Handles firing the `onChange` prop function any time the state changes.
+   * @param {Object} nextProps
+   * @param {Object} nextState
+   */
+  componentWillUpdate(nextProps, nextState) {
+    const viewId = this.state.view && this.state.view.id;
+    const nextViewId = nextState.view && nextState.view.id;
+
+    if (nextViewId != viewId) {
+      const {account, property, view} = nextState;
+      this.props.onChange({account, property, view});
+    }
+  }
+
+  /** @return {Object} The React component. */
+  render() {
+    if (this.state.hasNoViews) {
+      return (
+        <p>Oops, you don't appear to have any Google Analytics accounts, you
```

Comment:
Side note. Have you tested no GA accounts lately to make sure it's handled correctly?